tools/warnings_count/check_new_warnings.sh

```diff
@@ -6,11 +6,29 @@ set -e
 DIR="$(dirname "$(readlink -f "$0")")"
 cd "$DIR/../.."
 
+remove_ansi_codes () {
+    perl -pe '
+    s/\e\[[\x30-\x3f]*[\x20-\x2f]*[\x40-\x7e]//g;
+    s/\e[PX^_].*?\e\\//g;
+    s/\e\][^\a]*(?:\a|\e\\)//g;
+    s/\e[\[\]A-Z\\^_@]//g;' $1
+}
+
+make_warnings () {
+    make $1 2> >(tee tools/warnings_count/warnings_temp.txt) \
+    && remove_ansi_codes tools/warnings_count/warnings_temp.txt > tools/warnings_count/warnings_$2_new.txt \
+    && rm tools/warnings_count/warnings_temp.txt
+}
+
+compare_warnings () {
+    python3 tools/warnings_count/compare_warnings.py tools/warnings_count/warnings_$1_current.txt tools/warnings_count/warnings_$1_new.txt    
+}
+
 make distclean
-make setup 2> tools/warnings_count/warnings_setup_new.txt
-make disasm 2> tools/warnings_count/warnings_disasm_new.txt
-make all 2> tools/warnings_count/warnings_build_new.txt
+make_warnings setup setup
+make_warnings disasm disasm
+make_warnings all build
 
```

Comment:
```suggestion
make diff-init

```
Not 100% sure if this is necessary or not, but I thought check_new_warnings.sh was running a full `make init`, and it seems like it is doing all the pieces of it except for `make diff-init`. I presume this means diff.py will not work after running it? Unless `make distclean` doesn't actually remove the diff.py stuff.

Reply:
`distclean` does remove `expected`. The point is not to run `init`, the point is to test all the stages that can produce warnings (that Jenkins tests for). Running `make diff-init` separately after will just copy everything to the expected folder (assuming a successful build), so I don't really see much point in adding more time on to what is already a very time-consuming script to run for most people, who are used to using `-jN`.

Reply:
Fair, that makes sense. Though I can't imagine a simply copy operation of ~100 MB of files takes hardly any time at all. But it does make sense to conserve the number of disk writes at the very least, especially if check_new_warnings is being run multiple times in a row. Probably a question for another PR, but I wonder if we couldn't just create symlinks in `expected` to the files in repo root to avoid having to copy them all the time.

Reply:
The whole point of `expected` is to remain untouched and level with master (or a known good build) so that you can diff against it, though!

Reply:
Ah I see. I wonder if it would maybe make sense to have two makefile targets for building the roms. One that names the output file as what diff.py expects for a "known good build", and one that's named something else so that it can be used as "scratch" (run periodically to check matching). That might be a whole new build system PR at that point though and probably not worth it, just a musing.